examples/run_largecyc_CPMG.py

```diff
@@ -0,0 +1,224 @@
+    print(
+        "Based on that, and the gamma_eff_MHz_G you have in your .ini file, I'm setting the field to %f"
+        % field_G
+    )
+    with xepr() as x:
+        assert field_G < 3700, "are you crazy??? field is too high!"
+        assert field_G > 3300, "are you crazy?? field is too low!"
+        field_G = x.set_field(field_G)
+        print("field set to ", field_G)
+# }}}
+# {{{set phase cycling
+# NOTE: The overall phase and the 90-180 phase difference are phase cycled
+# in a nested way
+# Δp₁ l + (Δp₂)(m) + (Δp₁+Δp₂+Δp₃)(n) = (Δp₁)(l+n) + (Δp₂)(m+n) + (Δp₃)(n)
+ph1 = r_[0, 1, 2, 3]
+ph2 = r_[0, 1, 2, 3]
+ph_overall = r_[0, 1, 2, 3]
+# the following puts ph1 on the outside, which I would not have expected
+ph1_cyc = array([(l + n) % 4 for l in ph1 for m in ph2 for n in ph_overall])
+ph2_cyc = array([(m + n) % 4 for l in ph1 for m in ph2 for n in ph_overall])
+ph3_cyc = array([(n) % 4 for l in ph1 for m in ph2 for n in ph_overall])
+nPhaseSteps = 4**3
+# }}}
+# {{{ calibrate pulse lengths
+# NOTE: This is done inside the run_spin_echo rather than in the example
+# but to keep the generic function more robust we do it outside of the ppg
+prog_p90_us = prog_plen(config_dict["p90_us"])
+prog_p180_us = prog_plen(2 * config_dict["p90_us"])
+# }}}
+# {{{ calculate symmetric tau
+# NOTE: here the tau_us is defined as the evolution time from the start of
+# excitation (*during the pulse*) through to the start of the 180 pulse
+marker_us = 1.0  # the marker takes 1 us
+config_dict["tau_us"] = (
+    2 * config_dict["deadtime_us"] + 1e3 * config_dict["echo_acq_ms"]
+) / 2
+assert (
+    config_dict["tau_us"]
+    > 2 * prog_p90_us / pi + marker_us + config_dict["deblank_us"]
+)
+assert config_dict["deadtime_us"] > config_dict["deblank_us"] + 2 * marker_us
+print(
+    "If you are measuring on a scope, the time from the start (or end) of one 180 pulse to the next should be %0.1f us"
+    % (
+        2 * config_dict["deadtime_us"]
+        + 1e3 * config_dict["echo_acq_ms"]
+        + prog_p180_us
+    )
+)
+# }}}
+# {{{check total points
+total_pts = nPoints * nPhaseSteps * config_dict["nEchoes"]
+assert total_pts < 2**14, (
+    "You are trying to acquire %d points (too many points) -- either change SW or acq time so nPoints x nPhaseSteps is less than 16384"
+    % total_pts
+)
+# }}}
+# {{{ acquire CPMG
+# NOTE: Number of segments is nEchoes * nPhaseSteps
+data = generic(
+    ppg_list=[
+        ("phase_reset", 1),
+        ("delay_TTL", config_dict["deblank_us"]),
+        ("pulse_TTL", prog_p90_us, "ph_cyc", ph1_cyc),
+        (
+            "delay",
+            config_dict["tau_us"]
+            - 2 * prog_p90_us / pi
+            - marker_us
+            - config_dict["deblank_us"],
+        ),
+        ("delay", marker_us),  # placeholder for marker
+        ("delay_TTL", config_dict["deblank_us"]),
+        ("pulse_TTL", prog_p180_us, "ph_cyc", ph3_cyc),
```

Comment:
shouldn't this be ph2_cyc?

Reply:
yes correct, please change

Reply:
ph2_cyc also makes more sense to me comparing with planned pulse program